verifiers/internal/gha/slsaprovenance/v0.2/provenance.go

```diff
@@ -101,3 +127,94 @@ func (prov *ProvenanceV02) GetInputs() (map[string]interface{}, error) {
 	}
 	return pyldInputs, nil
 }
+
+func getBranchForTag(prov *ProvenanceV02) (string, error) {
+	// First try the base_ref.
+	environment, ok := prov.Predicate.Invocation.Environment.(map[string]interface{})
+	if !ok {
+		return "", fmt.Errorf("%w: %s", serrors.ErrorInvalidDssePayload, "parameters type")
+	}
+
+	baseRef, err := utils.GetAsString(environment, "github_base_ref")
+	if err != nil {
+		return "", err
+	}
+
+	// This `base_ref` seems to always be "".
+	if baseRef != "" {
+		return baseRef, nil
+	}
+
+	// Look at the event payload instead.
+	eventName, err := utils.GetAsString(environment, "github_event_name")
+	if err != nil {
+		return "", err
+	}
+
+	payload, err := getEventPayload(environment)
+	if err != nil {
+		return "", err
+	}
+
+	// We don't do that for all triggers because the payload
+	// is event-specific. Only `push` events seem to have a `base_ref`, and
+	// `release` events specify a branch in `target_commitish`.
+	switch eventName {
+	case "push":
+		value, err := getAsAny(payload, "base_ref")
+		if err != nil {
+			return "", err
+		}
+
+		// The `base_ref` field may be nil if the build was from
+		// a specific commit rather than a branch.
+		v, ok := value.(string)
+		if !ok {
+			return "", nil
+		}
+		return v, nil
+	case "release":
+		// For a release event, we look for release.target_commitish.
+		releasePayload, ok := payload["release"]
+		if !ok {
+			return "", fmt.Errorf("%w: %s", serrors.ErrorInvalidDssePayload, "release absent from payload")
+		}
+
+		release, ok := releasePayload.(map[string]interface{})
+		if !ok {
+			return "", fmt.Errorf("%w: %s", serrors.ErrorInvalidDssePayload, "parameters type releasePayload")
+		}
+
+		branch, err := utils.GetAsString(release, "target_commitish")
+		if err != nil {
+			return "", fmt.Errorf("%w: %s", err, "target_commitish not present")
+		}
+
+		return "refs/heads/" + branch, nil
+	default:
+		return "", nil
+	}
+}
+
+func getAsAny(environment map[string]any, field string) (any, error) {
+	value, ok := environment[field]
+	if !ok {
+		return "", fmt.Errorf("%w: %s", serrors.ErrorInvalidDssePayload,
+			fmt.Sprintf("environment type for %s", field))
+	}
+	return value, nil
+}
+
+func getEventPayload(environment map[string]interface{}) (map[string]interface{}, error) {
+	eventPayload, ok := environment["github_event_payload"]
+	if !ok {
+		return nil, fmt.Errorf("%w: %s", serrors.ErrorInvalidDssePayload, "parameters type event payload")
+	}
+
+	payload, ok := eventPayload.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("%w: %s", serrors.ErrorInvalidDssePayload, "parameters type payload")
+	}
```

Comment:
nit: Might as well use `any`
```suggestion
func getEventPayload(environment map[string]any) (map[string]any, error) {
	eventPayload, ok := environment["github_event_payload"]
	if !ok {
		return nil, fmt.Errorf("%w: %s", serrors.ErrorInvalidDssePayload, "parameters type event payload")
	}

	payload, ok := eventPayload.(map[string]any)
	if !ok {
		return nil, fmt.Errorf("%w: %s", serrors.ErrorInvalidDssePayload, "parameters type payload")
	}
```